pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -123,7 +124,7 @@ public static double get(OperationMetricKey key, ASTMethodOrConstructorDeclarati
      * @return The value of the metric, or {@code Double.NaN} if the value couln't be computed or {@code option} is
      * {@code null}
      */
-    public static double get(OperationMetricKey key, ASTAnyTypeDeclaration node, ResultOption option) {
+    public static double get(MetricKey<OperationMetric> key, ASTAnyTypeDeclaration node, ResultOption option) {
```

Comment:
I don't see how an operation metric and class metric could use the same node. ASTAnyTypeDeclaration and ASTMethodOrConstructorDeclaration are unrelated types, and the compiler type checks before it erases the type arguments. Perhaps the only instance where I could see a problem happening would be using reflection to call these methods, because the type argument would have been erased, but then that's highly unlikely.